atom/common/api/atom_bindings.cc

```diff
@@ -26,6 +28,12 @@ void Crash() {
   static_cast<DummyClass*>(NULL)->crash = true;
 }
 
+void Hang() {
+  // Temporarily access the current thread.
+  base::ThreadRestrictions::ScopedAllowSingleton allow_singleton;
+  base::WaitableEvent(false, false).Wait();
```

Comment:
if the call is gonna be made on the UI thread , accessing the current thread is the right way ? thats why went with [ScopedAllowSingleton](https://code.google.com/p/chromium/codesearch#chromium/src/base/threading/thread_restrictions.h&q=threadrestrictions&sq=package:chromium&l=121) though it states `doing it is almost always incorrect` .